### PR TITLE
Feat/386 only one pair

### PR DESCRIPTION
**Pull request by SushiRoll53**
- Check if screens have exact same name & same orientations, if they do append _n, to their names
- Iterating through the project just once

Let me know if you need the sketch file to test or you can always duplicate a screen to test

### CODE REVIEW

File: lib/interpret_and_optimize/helpers/pb_project.dart [hunk 1]
```diff
@@ -25,4 +25,40 @@ class PBProject {
...
+      var nextTree = forest[nextCounter];
+      if (currentTree.rootNode.name.compareTo(nextTree.rootNode.name) == 0) {
+        if (currentTree.data.orientation == nextTree.data.orientation) {
+          forest[nextCounter].rootNode.name =
+              forest[nextCounter].rootNode.name + '_$nameCounter';
+
+          nameCounter++;
+          nextCounter++;
+        } else {
+          currentCounter++;
```
Comment (ivan-015) [inline lib/interpret_and_optimize/helpers/pb_project.dart]: Rather than only increasing `currentCounter` by one, could we do something like this?
```
currentCounter = nextCounter;
nextCounter++;
```
In the case that there are no repeating names, currentCounter would simply increase by one. However, in the case that there are repeated names this would let us skip the trees we've already renamed rather than redundantly traversing them. 🤔 Let me know if you see flaws in this approach.

This would also apply for the else statement in `line 55`

Comment (SushiRoll53) [reply lib/interpret_and_optimize/helpers/pb_project.dart]: You are right, this can work, so would this get rid of the `if` statements on line 51 & 57??
They look like the following:
``` 
if (currentCounter >= nextCounter) {
          nextCounter++;
        }
```

Comment (SushiRoll53) [reply lib/interpret_and_optimize/helpers/pb_project.dart]: Follow up on that, yup, they are no longer needed

**Review by ivan-015 — APPROVED** [review]
LGTM!

**Comment by x64Eddie** [comment]
LGTM

**Review by x64Eddie — APPROVED** [review]
